src/main/ApplicationUtils.cpp

```diff
@@ -130,6 +195,35 @@ runWithConfig(Config cfg, std::optional<CatchupConfiguration> cc)
     return 0;
 }
 
+bool
+applyBucketsForLCL(Application& app)
+{
+    auto has = app.getLedgerManager().getLastClosedLedgerHAS();
+    auto lclHash =
+        app.getPersistentState().getState(PersistentState::kLastClosedLedger);
+
+    // As the local HAS might have merges in progress, let
+    // `prepareForPublish` convert it into a "valid historical HAS".
+    has.prepareForPublish(app);
```

Comment:
Another "carry over".

Is this `has.prepareForPublish(app);` really needed?
The current ledger state should never depend on merges in progress (when we close a ledger we have already computed the current ledger) - I think the only thing that this call is going to do is potentially cause a stall when restarting because it will perform merges that it doesn't need to do.

Reply:
This is currently needed since ApplyBucketsWork is designed to apply buckets from an untrusted source, so it verifies validity of the HAS. That being said, I think the check could be moved out of ApplyBucketsWork into catchup. I opened https://github.com/stellar/stellar-core/issues/3044 to track this.